src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -1197,5 +1322,114 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             true /*canCancelQueueItem*/, nullptr /*queueItem*/, package, options, std::move(callerProcessInfoString));
     }
 
+    winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Management::Deployment::AddPackageCatalogResult, double> PackageManager::AddPackageCatalogAsync(winrt::Microsoft::Management::Deployment::AddPackageCatalogOptions options)
+    {
+        LogStartupIfApplicable();
+
+        // options must be set.
+        if (!options)
+        {
+            co_return GetAddPackageCatalogResult(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS);

```

Comment:
Returning an object with an error code is the pattern used by other APIs in PackageManager. For example, WINGET_RETURN_DOWNLOAD_RESULT_HR_IF(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS, !package) appears to throw an error code directly, but it actually creates an object with an error status behind the scenes.

I can implement the suggested change, but it might not be consistent with the existing pattern used by other APIs.

Reply:
To simplify things, I'm considering moving this logic inside a try block and using THROW_HR_* constructs to throw invalid arguments. This way, the result object construction will be automatically handled at the point of return status in the following catch block.

Reply:
I have implemented the suggested PR feedback after reconsidering. Thanks, John, for explaining how this change aligns with the API Design guidelines and why it is beneficial.